src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -327,9 +327,27 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
     val stagingBucketName = generateUniqueBucketName("leostaging-"+clusterName.value)
 
     val googleFuture = for {
+
       // Validate that the Jupyter extension URI and Jupyter user script URI are valid URIs and reference real GCS objects
       _ <- validateBucketObjectUri(userEmail, googleProject, clusterRequest.jupyterExtensionUri)
       _ <- validateBucketObjectUri(userEmail, googleProject, clusterRequest.jupyterUserScriptUri)
+
+      _ <- clusterRequest.userJupyterExtensionConfig match {
+        case Some(x) => Future.traverse(x.nbExtensions)(x => validateBucketObjectUri(userEmail, googleProject, parseGcsPath(x._2).toOption))
+        case None => Future.successful(())
+      }
+
+      _ <- clusterRequest.userJupyterExtensionConfig match {
+        case Some(x) => Future.traverse((x.serverExtensions).filterKeys(_.startsWith("gs://")))(x => validateBucketObjectUri(userEmail, googleProject, parseGcsPath(x._2).toOption))
```

Comment:
Is `.filterKeys` right? I thought the keys were names, and the validates were gs:// paths.

Reply:
What if we did this:
```
_ <- clusterRequest.userJupyterExtensionConfig match {
  case Some(config) =>
    val extensionsToValidate = (config.nbExtensions.values ++ config.serverExtensions.values ++ config.combinedExtensions.values).filter(_.startsWith("gs://"))
    Future.traverse(extensionsToValidate)(x => validateBucketObjectUri(userEmail, googleProject, x))
   case None => Future.successful(())
}
```
and made `validateBucketObjectUri` take a `String` instead of a `Option[GcsPath]`. Then `validateBucketObjectUri` could do both `parseGcsPath()` _and_ `storageDAO.objectExists()` and throw an exception if either one fails.